index.js

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env node
+
+var program = require('commander');
+var colors = require('colors/safe');
+const RandExp = require('randexp');
+
+program
+    .version('1.0.0', '-v, --version')
+    .option('-l, --length <length>', 'password length. Minimal length is 8 characters, default length is 14 characters. Cannot be less than uppercase + digits + special', 14)
+    .option('-u, --uppercase <uppercase>', 'minimal number of uppercase characters. Default is 1. Cannot be greater than length – digits - special', 1)
+    .option('-d, --digits <digits>', 'minimal number of digits Default is 1. Cannot be greater than length – uppercase - special', 1)
+    .option('-s, --special <special>', 'minimal number of special characters. Default is 1. Cannot be greater than length – uppercase - digits', 1)
+    .parse(process.argv);
+
+
+/**
+ * Validation
+ * @param l - password length
+ * @param u - uppercase characters
+ * @param d - digits
+ * @param s - special characters
+ * @returns {Array}
+ */
+function lengthValidation(l, u, d, s) {
+    let errors = [];
+
+    if (l < (u + d + s)) {
+        errors.push('-l, --length – Cannot be less than uppercase (' + program.uppercase + ') + digits (' + program.digits + ') + special (' + program.special + ')');
+    }
+
+    if (u > (l - d - s)) {
+        errors.push('-u, --uppercase – Cannot be greater than length (' + program.uppercase + ') – digits (' + program.digits + ') + special (' + program.special + ')');
+    }
+
+    if (d > (l - u - s)) {
+        errors.push('-d, --digits – Cannot be greater than length (' + program.digits + ') – uppercase (' + program.uppercase + ') - special (' + program.special + ')');
+    }
+
+    if (s > (l - u - d)) {
+        errors.push('-s, --special – Cannot be greater than length (' + program.special + ') – uppercase (' + program.uppercase + ') - digits (' + program.digits + ')');
+    }
+
+    return errors;
+}
+
+/**
+ *
+ * @param number
+ * @param string
+ * @returns {boolean}
+ */
+function checkLength(number, string) {
+    return (typeof number !== 'number' || number.toString().length !== string.length);
+}
+
+/**
+ *
+ * @param l length
+ * @param u uppercase
+ * @param d digits
+ * @param s special
+ * @returns {Array}
+ */
+function validation(l, u, d, s) {
+    let errors = lengthValidation(l, u, d, s);
+
+    if (l < 8 || checkLength(l, program.length)) {
+        errors.push('Minimal password length is 8 characters');
+    }
+
+    if (u < 1 || checkLength(u, program.uppercase)) {
+        errors.push('At least 1 upper-case character');
+    }
+
+    if (d < 1 || checkLength(d, program.digits)) {
+        errors.push('At least 1 digit');
+    }
+
+    if (s < 1 || checkLength(s, program.special)) {
+        errors.push('At least 1 special character');
+    }
+
+    return errors;
+}
+
+/**
+ *
+ * @param a
+ * @returns {*}
+ */
```

Comment:
this `jsdoc` (as well as all above) is useless. `jsdoc` should describe input's and output's data types at least.